docs/content/concepts/assets/graph-backed-assets.mdx

```diff
@@ -0,0 +1,208 @@
+
+If the underlying computation is sufficiently flexible to selectively output a subset of assets, a graph-backed asset can be subsetted. For example, let’s say we wanted to define a graph-backed asset with the structure depicted in the image below. In this case, we want to independently materialize `foo_asset` and `baz_asset`.
+
+<Image
+alt="Graph-backed asset"
+src="/images/concepts/assets/graph-backed-asset.png"
+width={1568}
+height={1824}
+/>
+
+In order to selectively output an asset from a graph-backed asset, Dagster will run each op that is a dependency of the outputted asset. In the example, if we wanted to selectively materialize `foo_asset`, Dagster would run `foo` and `bar`. If we wanted to selectively materialize `baz_asset`, Dagster would run all three ops (`foo`, `bar`, and `baz`).
+
+Because the `foo` op yields an asset output (`foo_asset`) and is an upstream dependency of another asset generated from the graph (`baz_asset`), we need to structure `foo` to selectively return outputs depending on the asset subset selected for execution. We can do this by defining `foo` to have optional outputs that are yielded conditionally. Dagster provides a `context.selected_output_names` object on the op context that will return the outputs necessary to generate the asset subset.
+
+During execution, if we select just `baz_asset` for materialization, the below implementation of `foo` will return `{"foo_2"}` for `context.selected_output_names`, preventing `foo_asset` from being materialized.
+
+```python file=/concepts/assets/subset_graph_backed_asset.py startafter=start_graph_backed_asset_foo endbefore=end_graph_backed_asset_foo
+@op(out={"foo_1": Out(is_required=False), "foo_2": Out(is_required=False)})
+def foo(context, bar_1):
+    # Selectively returns outputs based on selected assets
+    if "foo_1" in context.selected_output_names:
+        yield Output(bar_1 + 1, output_name="foo_1")
+    if "foo_2" in context.selected_output_names:
+        yield Output(bar_1 + 2, output_name="foo_2")
+```
+
+Because Dagster flattens each graph into a flat input/output mapping between ops under the hood, any op that produces an output of the graph must be structured to yield its outputs optionally, enabling the outputs to be returned independently.
+
+In the example, `foo` and `baz` produce outputs of `my_graph`. Subsequently, their outputs need to be yielded optionally. Because `foo` yields multiple outputs, we must structure our code to conditionally yield its outputs in the code snippet above. However, because `baz` only yields a singular output, Dagster will only run `baz` when its asset output `baz_asset` is selected. So, we don’t have to structure `baz` to return an optional output. Because `bar` does not yield any outputs that are returned from `my_graph`, its outputs do not have to be selectively returned.
```

Comment:
This reads like there's a word missing here, so I guessed at the right one - feel free to toss this if it's not correct. I'm also suggesting adding a line break to visually break this up. I had to re-read this a few times to fully get it, as it felt really dense.

```suggestion
In the example, `foo` and `baz` produce outputs of `my_graph`. Subsequently, their outputs need to be yielded optionally. Because `foo` yields multiple outputs, we must structure our code to conditionally yield its outputs like in the code snippet above.

However, because `baz` only yields a singular output, Dagster will only run `baz` when its asset output `baz_asset` is selected. So, we don’t have to structure `baz` to return an optional output. Because `bar` does not yield any outputs that are returned from `my_graph`, its outputs do not have to be selectively returned.
```